### PR TITLE
[feature-wip](BE http)Support BE http service with brpc

### LINKED ISSUE #16087: [Enhancement] Support BE http service with brpc.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Doris BE uses `libevent` to support http service currently, which is planned to be replaced by `brpc` step by step in the future.

### Solution

Doris BE has `webserver_port` for http service and `brpc_port` for internal service. The brpc-based http service will share the `brpc_port` with internal service, and `webserver_port` will be kept for libevent-based http service at the current stage.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Hastyshell**
# Proposed changes

Issue Number: close #16087

## Problem summary

Doris BE uses `libevent` to support http service currently, which is planned to be replaced by `brpc` step by step in the future.

Doris BE has `webserver_port` for http service and `brpc_port` for internal service. The brpc-based http service will share the `brpc_port` with internal service, and `webserver_port` will be kept for libevent-based http service at the current stage.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.58 seconds
 load time: 490 seconds
 storage size: 17122274743 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230120102058_clickbench_pr_84689.html

**Comment by morningman** [comment]
Is there a way to keep http port unchanged but using brpc implemented http server?

**Comment by TangSiyang2001** [comment]
> Is there a way to keep http port unchanged but using brpc implemented http server?

Yes, doris has now got a brpc port for internal service and has established a brpc server. Becase brpc server supports several services ,we can add the new http service to this server to share the brpc port, and the original http port is completely unaffected.

**Comment by yangzhg** [comment]
@xiaojunjie pay attention to his pr, this maybe conflict with be support https/auth

**Comment by yongjinhou** [comment]
Didn't implement StreamLoadAction?

**Comment by yongjinhou** [comment]
Didn't implement StreamLoadAction?

**Comment by TangSiyang2001** [comment]
> Didn't implement StreamLoadAction?

Yes, brpc does not support continuous reception with http, refer to. After discussing with @dataroaring, we decided to implement it in another way, and the implementation will be in another pr, so that @xiaojunjie is able to start https works smoothly.  
Currently, I am focusing on regression-test and will make it completed as soon as possible.

**Comment by yongjinhou** [comment]
No MonitorAction in add_brpc_http_service()?

**Comment by TangSiyang2001** [comment]
> 

Cause no MonitorAction is registered to the original webserver, it seems redundant.

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p1

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

File: be/src/http/brpc/action/tablets_distribution_action.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                return;
+            }
+        }
+        on_succ_json(cntl,
+                     get_tablets_distribution_group_by_partition(cntl, partition_id).ToString());
+        return;
+    }
+    LOG(WARNING) << "invalid argument. group_by:" << *req_group_method;
+    Status status = Status::InternalError(strings::Substitute("invalid argument: group_by"));
+    std::string status_result = status.to_json();
```
Comment (platoneko) [inline be/src/http/brpc/action/tablets_distribution_action.cpp:109]: `strings::Substitute` seems redundant

Comment (TangSiyang2001) [reply be/src/http/brpc/action/tablets_distribution_action.cpp:109]: OK, I'll modify it.

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run build

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run p0

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run compile

**Comment by TangSiyang2001** [comment]
run p0

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**